ts/packages/node/src/stores/LaunchDarkly.ts

```diff
@@ -0,0 +1,173 @@
+import { Config, ConfigStore, ConfigStoreQuery } from '@configu/ts';
+import axios, { Axios } from 'axios';
+import assert from 'node:assert';
+import * as fs from 'fs';
+
+export type LaunchDarklyConfigStoreConfigurations = { apitoken: string; defaultproject: string } | string;
+
+export class LaunchDarklyConfigStore extends ConfigStore {
+  private client: Axios;
+  private readonly projectKey: string;
+  private environments: string[];
+
+  constructor(configurations: LaunchDarklyConfigStoreConfigurations) {
+    super('launch-darkly');
+    const configs =
+      typeof configurations === 'string'
+        ? JSON.parse(fs.readFileSync('ldc.json').toString())[configurations]
+        : configurations;
+    this.projectKey = configs.defaultproject;
+    this.client = axios.create({
+      baseURL: `${configs.server ? configs.server : 'https://app.launchdarkly.com'}/api/v2`,
+      headers: {
+        Authorization: configs.apitoken,
+      },
+      responseType: 'json',
+    });
+    this.environments = [];
+  }
+
+  private async patchUpdate(config: Config, patchData: Record<string, any>) {
+    const { data } = await this.client.patch(`/flags/${this.projectKey}/${config.key}`, patchData, {
+      headers: { 'Content-Type': 'application/json; domain-model=launchdarkly.semanticpatch' },
+    });
+    return data;
+  }
+
+  private async updateEnvironments() {
+    const { data: environments } = await this.client.get(`/projects/${this.projectKey}/environments`);
+    this.environments = [...this.environments, ...environments.items.map((env: any) => env.key)];
+  }
+
+  private async getFeatureFlag(config: Config) {
+    const { data: featureFlag } = await this.client.get(`/flags/${this.projectKey}/${config.key}`);
+    return featureFlag;
+  }
+
+  private async createEnvironment(config: Config) {
+    await this.client.post(`/projects/${this.projectKey}/environments`, {
+      color: 'FFFFFF',
+      key: config.set,
+      name: config.set,
+    });
+    this.environments = [...this.environments, config.set];
+  }
+
+  private async createFeatureFlag(config: Config) {
+    const createData = {
+      clientSideAvailability: {
+        usingEnvironmentId: true,
+        usingMobileKey: true,
+      },
+      key: config.key,
+      name: config.key,
+      variations: [{ value: config.value }, { value: '' }],
+    };
+    const { data: featureFlag } = await this.client.post(`/flags/${this.projectKey}`, createData);
+    return featureFlag;
+  }
+
+  private async createVariation(config: Config) {
+    const createVariationData = {
+      instructions: [{ kind: 'addVariation', value: config.value }],
+    };
+    const data = await this.patchUpdate(config, createVariationData);
+    return data.variations.find((variation: any) => variation.value === config.value);
+  }
```

Comment:
This will only work if the feature flag is of type string. If the user has existing flags that are not of type string, this will fail with the following error:
<img width="1007" alt="image" src="https://github.com/configu/configu/assets/47181182/98a985c8-9e15-470e-b2c2-705e0aae04f3">

We need to rethink this logic, we can't simply make this assumption because it will either make this store unusable or force users to recreate their feature flags with the string type

Reply:
That's a good point.. I think i can solve this with JSON.parse/stringfy on our end